### ISSUE TITLE
License Precision: GPLv3 only or extended?

### ISSUE THREAD

**Issue by Sigmanificient** [cococry/Ragnar#31]
Hi, I am currently trying [to ship Ragnar into the nixpkgs](https://github.com/NixOS/nixpkgs/pull/247909).

My proposition for a derivation as been reviewed, and within the changes requested lies a question about the license.
Nix distinguishes between gpl "only" and "plus", and we need to know in which this project is.

> *The ambiguous `gpl3` attribute is sorta-outdated, we prefer clarifying if it's `gpl3Only` (without the "or (at your option) any later version" bit) or `gpl3Plus` (with it).*
> 
> *[The license text](https://github.com/cococry/Ragnar/blob/4b485b78fefb7539b17b1115b3f75eb0c481ea55/LICENSE#L625-L632) expects the developer to include the GPL notice at the start of their code files. If the aforementioned line is included in these, then [the license says](https://github.com/cococry/Ragnar/blob/4b485b78fefb7539b17b1115b3f75eb0c481ea55/LICENSE#L570-L577) that code may be "upgraded" to any later GPL version.*

**Comment by cococry** [comment]
Ragnar is using the GNU general public v3.0 license

**Comment by Sigmanificient** [comment]
Seems on the LICENSE file that you are set the license as Gpl3Plus
![image](https://github.com/cococry/Ragnar/assets/53050011/0f9295b9-4943-45de-b7d9-7025fe67fd7b)

**Comment by OPNA2608** [comment]
> Seems on the LICENSE file that you are set the license as Gpl3Plus

No, the `LICENSE` file is a verbatim text copy of the GPL-3 license. The shown paragraph is the section of the license text that gives an example of the notice a developer is supposed to put at the start of all their source code files. The inclusion of the "any later version" line there is just an example [(and what the FSF recommends)](https://www.gnu.org/licenses/gpl-faq.html#VersionThreeOrLater).

https://github.com/cococry/Ragnar/blob/4b485b78fefb7539b17b1115b3f75eb0c481ea55/LICENSE#L625-L632

---

> Ragnar is using the GNU general public v3.0 license

My understanding of this is that you intend for this to be GPLv3-Only, without the optional "any later version" clause:
```
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, version 3 of the License.
```

As opposed to GPLv3-Plus, which would allow users of applying a hypothetical GPLv4 if they want:
```
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
```

Is this correct?
